src/Network/GRPC/MQTT/Client.hs

```diff
@@ -221,3 +222,24 @@ grpcTimeout :: (MonadUnliftIO m) => TimeoutSeconds -> m (MQTTResult streamType r
 grpcTimeout timeLimit action = fromMaybe timeoutError <$> timeout (timeLimit * 1000000) action
  where
   timeoutError = GRPCResult $ ClientErrorResponse (ClientIOError GRPCIOTimeout)
+
+generateSessionId :: Generator -> IO Topic
+generateSessionId randGen = go 0
+ where
+  go :: Int -> IO Topic
+  go retries
```

Comment:
Why do you need to retry `mkTopic`? This kind of retry loop seems okay to me, but have you considered the `retry` package too?

Reply:
For the off chance that the generated session ID doesn't form a valid `Topic` (i.e. contains a '+' or '#'). But, I am right now realizing that the "nonce128**url**T" won't generate those characters... So I will take this out.